onnxruntime/contrib_ops/cpu/bert/decoder_masked_multihead_attention.cc

```diff
@@ -0,0 +1,482 @@
+                          head_size, v_head_size, v_hidden_size, attention_bias, context, cross_qk,
+                          parameters.past_sequence_length, true /* use_dmmha */);
+  }
+
+  // Self-attention, has_beams
+  return ApplyAttentionWithBeams(Q.GetMutable<Tensor>()->MutableData<T>(),
+                                 K.GetMutable<Tensor>()->MutableData<T>(),
+                                 V.GetMutable<Tensor>()->MutableData<T>(),
+                                 mask_index, past_key, past_value, output, present_key, present_value,
+                                 batch_size, parameters.past_sequence_length, parameters.max_sequence_length,
+                                 head_size, v_head_size, v_hidden_size, attention_bias, cache_indir, context,
+                                 beam_width_value, cross_qk);
+}
+
+template <typename T>
+Status DecoderMaskedMultiHeadAttention<T>::ApplyAttentionWithBeams(
+    const T* Q,
+    const T* K,
+    const T* V,
+    const Tensor* mask_index,
+    const Tensor* past_key,
+    const Tensor* past_value,
+    Tensor* output,
+    Tensor* present_key,
+    Tensor* present_value,
+    int batch_size,
+    int past_sequence_length,
+    int max_sequence_length,
+    int head_size,
+    int v_head_size,
+    int v_hidden_size,
+    const Tensor* attn_bias,
+    const Tensor* cache_indir,
+    OpKernelContext* context,
+    int beam_width,
+    Tensor* scaled_qk) const {
+  AllocatorPtr allocator;
+  ORT_RETURN_IF_ERROR(context->GetTempSpaceAllocator(&allocator));
+
+  auto* tp = context->GetOperatorThreadPool();
+
+  int total_sequence_length = past_sequence_length + 1;
+  size_t bytes = SafeInt<size_t>(batch_size) * num_heads_ * total_sequence_length * sizeof(T);
+  auto attention_probs = allocator->Alloc(bytes);
+  BufferUniquePtr scratch_buffer(attention_probs, BufferDeleter(allocator));
+
+  T* scaled_qk_data = (scaled_qk != nullptr) ? scaled_qk->MutableData<T>() : nullptr;
+
+  const int32_t* mask_index_data = mask_index != nullptr ? mask_index->Data<int32_t>() : nullptr;
+  const T* attn_bias_data = attn_bias != nullptr ? attn_bias->Data<T>() : nullptr;
+
+  ComputeAttentionProbsWithBeams(static_cast<T*>(attention_probs), Q, K, mask_index_data, batch_size,
+                                 past_sequence_length, max_sequence_length, head_size, past_key->Data<T>(),
+                                 present_key->MutableData<T>(), tp, attn_bias_data,
+                                 cache_indir->Data<int32_t>(), beam_width, scaled_qk_data);
+
+  // Compute the attentionScore * Value: out_tmp(B, N, 1, H_v) = attention_probs(B, N, 1, T) x V(B, N, T, H_v)
+  auto out_tmp_data = allocator->Alloc(SafeInt<size_t>(batch_size) * num_heads_ * v_head_size * sizeof(T));
+  BufferUniquePtr out_tmp_buffer(out_tmp_data, BufferDeleter(std::move(allocator)));
+
+  ComputeVxAttentionScoreWithBeams(output->MutableData<T>(), static_cast<T*>(out_tmp_data),
```

Comment:
Fix this: too few arguments to function call, expected 13, have 12

Reply:
Thanks, fixed. Local changes missed when pushing.